openedx/core/djangolib/blockstore_cache.py

```diff
@@ -168,7 +176,7 @@ def get_bundle_draft_files_cached(bundle_uuid, draft_name):
     get automatic cache invalidation when the draft is updated.
     """
     bundle_cache = BundleCache(bundle_uuid, draft_name)
-    cache_key = ('bundle_draft_files', )
+    cache_key = ('bundle_draft_files', draft_name)
```

Comment:
Hmm, you're right that the `draft_name` should be added to the key, but I would rather do it more generically via the line above (`bundle_cache = BundleCache(bundle_uuid, draft_name)` should incorporate the `draft_name` into the cache).

So change `_get_versioned_cache_key()` so that it passes `draft_name` into `_construct_versioned_cache_key` and change `_construct_versioned_cache_key` to append `:{draft_name}` to the key if draft_name is specified.

Reply:
+1